core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -1020,6 +1052,87 @@ public void releaseRegisterLease(long workerId) {
     mRegisterLeaseManager.releaseLease(workerId);
   }
 
+  @Override
+  public String getClusterId() {
+    try {
+      // ClusterId is generated by MetaMaster,
+      // MetaMaster is constructed later than BlockMaster,
+      // so wait for MetaMaster to generate and set ClusterId.
+      CommonUtils.waitFor(this + " getClusterId",
```

Comment:
`this + "getClusterId"` will concat strings every time this is called. So either put a constant string here or use String.format().